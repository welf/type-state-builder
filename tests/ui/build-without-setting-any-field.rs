use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct Person {
    #[builder(required)]
    name: String,

    #[builder(required)]
    age: u32,
}

fn main() {
    // This should fail: trying to build without setting any required fields
    let person = Person::builder().build();
}
