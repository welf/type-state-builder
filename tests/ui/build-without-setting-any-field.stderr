error[E0599]: no method named `build` found for struct `PersonBuilder_MissingName_MissingAge` in the current scope
  --> tests/ui/build-without-setting-any-field.rs:14:36
   |
3  | #[derive(TypeStateBuilder)]
   |          ---------------- method `build` not found for this struct
...
14 |     let person = Person::builder().build();
   |                                    ^^^^^ method not found in `PersonBuilder_MissingName_MissingAge`
